code/week8/main.cpp

```diff
@@ -0,0 +1,105 @@
+#include <iostream>
+#include <algorithm>
+#include <vector>
+
+int addTwo(int num) {
+    return num + 2;
+}
+
+int addThree(int num) {
+    return num + 3;
+}
+
+int add(int num1, int num2) {
+    return num1 + num2;
+}
+
+void printIntVec(std::vector<int> vec) {
+    std::cout << "printing vector " << std::endl;
+    for(std::vector<int>::iterator it = vec.begin();it != vec.end(); it++) {
+        std::cout << *it << ", ";
+    }
+    std::cout << "\n\n";
+}
+
+void printInt(int i) {
+    std::cout << "looking at " << i << std::endl;
+}
+
+class Parent {
+public:
+    virtual void do_something() {std::cout << "BASE" << std::endl;}
```

Comment:
can this print be changed to `parent` or the base class be renamed to `base`, it's a little weird to have multiple names for the same thing.